### DISCUSSION TITLE
Singlestore interoperability

### DISCUSSION THREAD

**Discussion by paulm17** [geldata/gel#3448] [Ideas]
Right now, edge does not support horizontal scaling, sharding, failover.  Basically HA.  

Yes, I know there are [promises] for something to come out.  But nothing conclusive, at least in my book.

That said.  Singlestore is available with all 3.  However, it's rooted in mysql.

Are there any plans for Singlestore support?  It would be a game changer in this space, if edgeDB supported it. 

Thanks

**Comment by elprans** [comment]
I replied in #3403 why targeting a different RDBMs is currently not an option.

> Right now, edge does not support horizontal scaling, sharding, failover. Basically HA.

EdgeDB natively supports Postgres HA and can itself run in clustered mode: https://www.edgedb.com/docs/reference/backend_ha

**Reply by paulm17** [reply]
> During backend failover, no frontend connections will be closed; instead, all incoming queries will fail with a retryable error until failover has completed successfully. 

It's barely HA.  This is no better than what I was running back in 2012!  

Hopefully this changes.  Thanks for the further clarification on why edgedb only supports postgres.

**Reply by elprans** [reply]
> It's barely HA. This is no better than what I was running back in 2012!

EdgeDB clients survive network errors, including during failover by _automatically_ retrying the operation.  If failover happens quickly, your app will not even notice it happened (aside from a bit of extra latency).  If that's not HA, I'm not sure what is.
